lib/ui/page/home/page/chat/controller.dart

```diff
@@ -2375,6 +2422,22 @@ class ChatController extends GetxController {
   }
 }
 
+/// This class is used to persist and restore scroll state, especially when navigating
+/// between chats or rebuilding the widget tree. It includes:
+/// This helps ensure a smooth user experience by restoring the user's position
+/// accurately in chat threads, even with large numbers of messages.
+class ChatScrollInfo {
+  final double scrollOffset;
+  final int index;
+  final bool offsetBasedOnBottom;
+
+  const ChatScrollInfo({
```

Comment:
1) Fields should have documentation.
2) Default constructor should be placed first according to the [Contribution Guide](https://github.com/team113/messenger/blob/main/CONTRIBUTING.md#classes-constructors-fields-and-methods-ordering) rules.